sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java

```diff
@@ -68,8 +67,22 @@ public XmlSensor(FileSystem fileSystem, CheckFactory checkFactory, FileLinesCont
       fileSystem.predicates().hasLanguage(Xml.KEY));
   }
 
-  private void computeLinesMeasures(SensorContext context, XmlFile xmlFile) {
-    LineCounter.analyse(context, fileLinesContextFactory, xmlFile);
+  @Override
+  public void execute(SensorContext context) {
+    Optional<RuleKey> parsingErrorKey = getParsingErrorKey();
+
+    for (InputFile inputFile : fileSystem.inputFiles(mainFilesPredicate)) {
+      XmlFile xmlFile = new XmlFile(inputFile, fileSystem);
+      try {
+        NewXmlFile newXmlFile = NewXmlFile.create(inputFile);
+        LineCounter.analyse(context, fileLinesContextFactory, newXmlFile);
+        runChecks(context, xmlFile);
```

Comment:
you should provide the `newXmlFile` here as well, otherwise we are going to recompute one at line 100.